src/grt/src/fastroute/src/maze.cpp

```diff
@@ -2130,34 +2130,99 @@ void FastRouteCore::mazeRouteMSMD(const int iter,
   v_cost_table_.clear();
 }
 
+std::vector<std::string> FastRouteCore::getNetsInCongestedEdge(int x,
```

Comment:
Looping all the routing for each edge could be rather slow if there are many congested edges.  It might be better to store  all the edges in a hash by x/y->nets and the loop the routing once to update the nets.